selftests/unit/test_job.py

```diff
@@ -0,0 +1,119 @@
+                pass
+        return found
+
+    def test_job_empty_suite(self):
+        args = argparse.Namespace()
+        empty_job = job.Job(args)
+        self.assertIsNone(empty_job.test_suite)
+
+    def test_job_empty_has_id(self):
+        args = argparse.Namespace()
+        empty_job = job.Job(args)
+        self.assertIsNotNone(empty_job.unique_id)
+
+    def test_job_test_suite_not_created(self):
+        args = argparse.Namespace()
+        myjob = job.Job(args)
+        self.assertIsNone(myjob.test_suite)
+
+    def test_job_create_test_suite_empty(self):
+        args = argparse.Namespace()
+        myjob = job.Job(args)
+        myjob.create_test_suite()
+        self.assertEqual(myjob.test_suite, [])
+
+    def test_job_create_test_suite_simple(self):
+        simple_tests_found = self._find_simple_test_candidates()
+        args = argparse.Namespace(url=simple_tests_found)
+        myjob = job.Job(args)
+        myjob.create_test_suite()
+        self.assertEqual(len(simple_tests_found), len(myjob.test_suite))
+
+    def test_job_pre_tests(self):
+        class JobFilterTime(job.Job):
+            def pre_tests(self):
+                filtered_test_suite = []
+                for test_factory in self.test_suite:
+                    if test_factory[0] is test.SimpleTest:
+                        if not test_factory[1].get('name', '').endswith('time'):
+                            filtered_test_suite.append(test_factory)
+                self.test_suite = filtered_test_suite
+                super(JobFilterTime, self).pre_tests()
+        simple_tests_found = self._find_simple_test_candidates()
+        args = argparse.Namespace(url=simple_tests_found)
+        myjob = JobFilterTime(args)
+        myjob.create_test_suite()
+        myjob.pre_tests()
+        self.assertLessEqual(len(myjob.test_suite), 1)
+
+    def test_job_run_tests(self):
+        simple_tests_found = self._find_simple_test_candidates(['true'])
+        args = argparse.Namespace(url=simple_tests_found)
+        myjob = job.Job(args)
+        myjob.create_test_suite()
+        self.assertEqual(myjob.run_tests(),
+                         exit_codes.AVOCADO_ALL_OK)
+
+    def test_job_post_tests(self):
+        class JobLogPost(job.Job):
+            def post_tests(self):
+                with open(os.path.join(self.logdir, "reversed_id"), "w") as f:
+                    f.write(self.unique_id[::-1])
+                super(JobLogPost, self).post_tests()
+        simple_tests_found = self._find_simple_test_candidates()
+        args = argparse.Namespace(url=simple_tests_found)
+        myjob = JobLogPost(args)
+        myjob.create_test_suite()
+        myjob.pre_tests()
+        myjob.run_tests()
+        myjob.post_tests()
+        self.assertEqual(myjob.unique_id[::-1],
+                         open(os.path.join(myjob.logdir, "reversed_id")).read())
+
+    @unittest.skip("Issue described at https://trello.com/c/qgSTIK0Y")
```

Comment:
@apahim I tried your suggestion of using something like:

```python
import json
from avocado.utils import download

...
...

def _is_issue_fixed():
    issue_url = 'https://trello.com/card/58079f5f3dea443d1800fe43/single-data-dir-get-tmp-dir-per-interpreter-breaks-multiple-jobs.json'
    issue = json.load(download.url_open(issue_url))
    if 'Done' in issue['actions'][0]['data']['list']['name']:
        return True
    return False
```

But while it worked while checking this card is not *Done*, it failed for cards on *Done*, such as:

https://trello.com/card/57f5e24994bcce1c223cfb60/bug-the-cannot-sudo-from-selftests-unit-test-utils-partition-py-most-certainly-does-not-work.json

So, I decided to track this is a checklist on the trello card of the issue:

https://trello.com/c/qgSTIK0Y/859-single-data-dir-get-tmp-dir-per-interpreter-breaks-multiple-jobs